### PR TITLE
[Sprint-2][Team-3][Task-1] - create automation script for NetworkSecurityGroup

### LINKED ISSUE #55: [Sprint-2][Team-3][Task-1] - create automation script for NetworkSecurityGroup
Use New-AzRespourceGroupDeployment with template.json and without parameters.json

### PR DESCRIPTION
**Pull request by OyunominJuno**

### CODE REVIEW

File: automation/network-security-group/create-nsg-v2.ps1 [hunk 1]
```diff
@@ -26,17 +28,26 @@ param(
 
     [Parameter(Mandatory=$True)]
     [string]
-    $Location,            #Resource group location
+    $LocationForNetworkSecurityGroup,     #Resource group location
 
     [Parameter(Mandatory=$True)]
     [string]
-    $NetworkSecurityGroupName  #Network Secutiry Name to create
+    $TemplateJsonFilePath  #Path of the template.json file, either absolute or relative
```
Comment (sunshinejen) [inline automation/network-security-group/create-nsg-v2.ps1:43]: What happens if a user enters incorrect file path?

**Review by sunshinejen — APPROVED** [review]
Clean and easy to read, commented well. Tested and verified that NSG was created on Azure. I am not sure if Toddy had requirements for naming conventions for NSG like nsc-nsg-dev-team3 for example, and how is user aware of this when entering the NSG name? Two small things to think about, otherwise good job.

File: automation/network-security-group/create-nsg-v2.ps1 (hunk 1 above)
Comment (OyunominJuno) [reply automation/network-security-group/create-nsg-v2.ps1:43]: oh very good question, i will look into it. thank you

Comment (OyunominJuno) [reply automation/network-security-group/create-nsg-v2.ps1:43]: i added a validation for template.json file path input

**Review by sunshinejen — APPROVED** [review]
Looks good!

**Review by elb-dev — CHANGES_REQUESTED** [review]
It works and I was able to create a network security group, but I am fairly sure that Toddy requested that we _not_ have to open up a web browser in order to log in. You can change that with the following code:
```
$Credential = Get-Credential
Connect-AzAccount -Credential $Credential -Tenant $TenantId -ServicePrincipal
```
Furthermore, I'm a little confused at the purpose behind having the template.json be something you punch in yourself. It would make more sense for that to be something that's optionally set by the user when they type in the command to run the script. Otherwise that could default to what's there.

File: automation/network-security-group/template.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "networkSecurityGroupName": {
+            "type": "string",
+            "metadata": {
+                "description": "This is the name of the network security group"
+            }
+        },
+        "location": {
```
Comment (asiagromasia) [inline automation/network-security-group/template.json]: adding allowedValues options here would limit user input making it more safe

**Comment by asiagromasia** [comment]
adding outputs in template file would help with referencing this resource

File: automation/network-security-group/template.json (hunk 2 above)
Comment (OyunominJuno) [reply automation/network-security-group/template.json]: i added this and tested it

**Comment by OyunominJuno** [comment]
@elb-dev I see what you mean but I specifically wanted to the script to be dynamic as possible. If the user wanted to create not only an NSG but other resources with different template files, my script works too. So I think I will leave this way. 
For the credentials, I am not able to log in with my username and password so I cannot use your way.

**Comment by OyunominJuno** [comment]
Changed the way of user signing into Azure, used Credentials as Derek suggested

**Review by elb-dev — APPROVED** [review]

**Pull request merged**